### PR TITLE
Clarify a11yTitle docs

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR clarifies what is added to the DOM when an a11yTitle prop is provided. The name of this prop is slightly deceiving because ARIA has attributes called `aria-label` and `title`. While our prop is called `a11yTitle`, it is actually affecting the `aria-label`. While it wouldn't be smart to change the name of our prop due to backwards compatibility, a clarification to the docs may be helpful to users.

#### Where should the reviewer start?
src/js/utils/prop-types.js

#### What testing has been done on this PR?
The tests were updated to reflect the change in wording.

#### Any background context you want to provide?
While our prop is called `a11yTitle`, it is actually affecting the `aria-label`. While it wouldn't be smart to change the name of our prop due to backwards compatibility, a clarification to the docs may be helpful to users.

#### What are the relevant issues?
#3517 

#### Do the grommet docs need to be updated?
Yes, this PR updates docs only.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, this is backwards compatible.

### CODE REVIEW

File: src/js/utils/prop-types.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { PropTypes } from 'react-desc';
 
 export const a11yTitlePropType = PropTypes.string.description(
-  'Custom title to be used by screen readers.',
+  `Custom label to be used by screen readers. When provided, an aria-label will
+   be added to the element.`,
```
Comment (ShimiSun) [inline src/js/utils/prop-types.js]: @ericsoderberghp or @alansouzati, was there a reason we avoided mentioning that `a11yTitle` is actually being translated to `aria-label` under the hood?

Comment (ericsoderberghp) [reply src/js/utils/prop-types.js]: I can't remember. :(

**Comment by ShimiSun** [comment]
@taysea I think some updated snapshots are missing here 
Screen Shot 2019-11-15 at 8 23 11 AM

**Comment by taysea** [comment]
Let me update those! @ShimiSun

**Comment by ShimiSun** [comment]
Thank you! 🎉

**Pull request merged**